src/open_producten/producttypen/tests/api/test_prijs.py

```diff
@@ -0,0 +1,432 @@
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(Prijs.objects.count(), 0)
+        self.assertEqual(PrijsOptie.objects.count(), 0)
+
+    def test_get_actuele_prijs_when_product_type_has_no_prijzen(self):
+        response = self.client.get("/api/v1/producttypen/actuele-prijzen/")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data,
+            [
+                {
+                    "id": str(self.product_type.id),
+                    "naam": self.product_type.naam,
+                    "upl_naam": self.product_type.uniforme_product_naam.naam,
+                    "upl_uri": self.product_type.uniforme_product_naam.uri,
+                    "actuele_prijs": None,
+                },
+            ],
+        )
+
+    def test_get_actuele_prijzen_when_product_type_only_has_prijs_in_future(self):
+        PrijsFactory.create(
+            product_type=self.product_type, actief_vanaf=datetime.date(2024, 2, 2)
+        )
+
+        response = self.client.get("/api/v1/producttypen/actuele-prijzen/")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data,
+            [
+                {
+                    "id": str(self.product_type.id),
+                    "naam": self.product_type.naam,
+                    "upl_naam": self.product_type.uniforme_product_naam.naam,
+                    "upl_uri": self.product_type.uniforme_product_naam.uri,
+                    "actuele_prijs": None,
+                },
+            ],
+        )
+
+    def test_get_actuele_prijzen_when_product_type_has_actuele_prijs(self):
+        prijs = PrijsFactory.create(
+            product_type=self.product_type,
+            actief_vanaf=datetime.date(2024, 1, 1),
+        )
+
+        optie = PrijsOptieFactory.create(prijs=prijs)
+
+        response = self.client.get("/api/v1/producttypen/actuele-prijzen/")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data,
+            [
+                {
+                    "id": str(self.product_type.id),
+                    "naam": self.product_type.naam,
+                    "upl_naam": self.product_type.uniforme_product_naam.naam,
+                    "upl_uri": self.product_type.uniforme_product_naam.uri,
+                    "actuele_prijs": {
+                        "id": str(prijs.id),
+                        "product_type_id": self.product_type.id,
+                        "actief_vanaf": "2024-01-01",
+                        "prijsopties": [
+                            {
+                                "bedrag": str(optie.bedrag),
+                                "beschrijving": optie.beschrijving,
+                                "id": str(optie.id),
+                            }
+                        ],
+                    },
+                },
+            ],
+        )
```

Comment:
This is what I mean by using the full result instead of the generator mentioned in other places BTW